clang/test/CodeGenSYCL/kernel-early-optimization-pipeline.cpp

```diff
@@ -22,4 +23,17 @@
 //
 // RUN: %clang_cc1 -O2 -fsycl-is-device -triple spir64-unknown-unknown %s -mdebug-pass Structure -emit-llvm -fno-sycl-early-optimizations -o /dev/null 2>&1 | FileCheck %s --check-prefix=CHECK-NOEARLYOPT
 // CHECK-NOEARLYOPT-NOT: ConstantMergePass1
+// CHECK-NOEARLYOPT-NOT: SYCLOptimizeBarriersPass
 // CHECK-NOEARLYOPT: SYCLMutatePrintfAddrspacePass
+
+// RUN: %clang_cc1 -O0 -fsycl-is-device -triple spir64-unknown-unknown %s -mdebug-pass Structure -emit-llvm -o /dev/null 2>&1 | FileCheck %s --check-prefix=CHECK-O0
+// CHECK-O0-NOT: SYCLOptimizeBarriersPass
+
+template <typename name, typename Func>
+void kernel(const Func &f) __attribute__((sycl_kernel)) {
+  f();
+}
+
+void bar() {
+  kernel<class MyKernel>([=]() {});
```

Comment:
Apparently non-empty source code is required to make the pass run. @Fznamznon please comment if you are okay with this change or if I should separate the barrier case from this test.

Reply:
I think it is fine in this file, I don't really get why it happens.